### DISCUSSION TITLE
Error after Laravel 9 Upgrade

### DISCUSSION THREAD

**Discussion by jersey-mike** [spatie/laravel-webhook-client#142] [Q&A] [answered]
Invalid route action: [App\Http\Controllers\Spatie\WebhookClient\Http\Controllers\WebhookController].

  at C:\laragon\www\***\vendor\laravel\framework\src\Illuminate\Routing\RouteAction.php:92
     88▕      */
     89▕     protected static function makeInvokable($action)
     90▕     {
     91▕         if (! method_exists($action, '__invoke')) {
  ➜  92▕             throw new UnexpectedValueException("Invalid route action: [{$action}].");
     93▕         }
     94▕
     95▕         return $action.'@__invoke';
     96▕     }

  1   C:\laragon\www\***\vendor\laravel\framework\src\Illuminate\Routing\RouteAction.php:47
      Illuminate\Routing\RouteAction::makeInvokable("App\Http\Controllers\Spatie\WebhookClient\Http\Controllers\WebhookController")

  2   C:\laragon\www\***\vendor\laravel\framework\src\Illuminate\Routing\Route.php:190
      Illuminate\Routing\RouteAction::parse("api", ["App\Http\Controllers\Spatie\WebhookClient\Http\Controllers\WebhookController", "App\Http\Controllers\Spatie\WebhookClient\Http\Controllers\WebhookController"])

**Comment by jeremyallebe** [comment]
Hi,

Did you find a workaround?

Thanks

**Reply by pocketnest-erik** [reply]
Need this solution as well.

**Comment by jersey-mike** [comment]
The workaround I found was to set up normal POST or GET routes that call the web hook controller.

**Comment by ankitsinghdalal** [comment, marked as answer]
You just need to change the namespace for the webhook routes. For example

`Route::group(['prefix' => 'v1/webhook', 'namespace' => '\Spatie\WebhookClient\Http\Controllers'], function () {
    Route::webhooks('some-route');
});`

This way it is working for me. Further you can do `php artisan route:list` to check if the webhook route is added successfully.
